angularjs-portal-frame/src/main/webapp/portal/notifications/controllers.js

```diff
@@ -4,44 +4,53 @@ define(['angular'], function(angular) {
 
   var app = angular.module('portal.notifications.controllers ', []);
 
-  app.controller('NotificationController', [ 'notificationsService', function(notificationsService){
-    var store = this;
-    store.notifications = [];
-    store.count = 0;
-    store.isEmpty = false;
-    store.notificationUrl = 'notifications';
-
-    notificationsService.getAllNotifications().then(function(data){
+  app.controller('NotificationController', [ '$scope','NOTIFICATION', 'notificationsService', function($scope, NOTIFICATION, notificationsService){
+    var successFn = function(data){
       //success state
-      store.count = data.length;
-      store.isEmpty = (store.count === 0);
-      if(store.isEmpty) {
-        store.status = "No notifications";
-      } else {
-        store.status = "You have notifications";
-      }
-      store.notifications = data;
-    }, function(data){
+      $scope.count = data.length;
+      $scope.isEmpty = ($scope.count === 0);
+      $scope.status = "You have "+ ($scope.isEmpty ? "no " : "") + "notifications";
+      $scope.notifications = data;
+    };
+
+    var errorFn = function(data){
       //error state (logging of error happens at service layer)
-      store.count = 0;
-      store.isEmpty = true;
-    });
+      $scope.count = 0;
+      $scope.isEmpty = true;
+    };
+
+
 
-    this.getClass = function getClass(index, notifications) {
+    $scope.getClass = function getClass(index, notifications) {
       return {
         unread : notifications[index].attributes == undefined || notifications[index].attributes.READ == undefined || !notifications[index].attributes.READ,
         read : notifications[index].attributes != undefined && notifications[index].attributes.READ  != undefined && notifications[index].attributes.READ
       };
     };
 
-    this.markRead = function markRead(index, notifications) {
+    $scope.markRead = function markRead(index, notifications) {
       if(notifications[index].attributes == undefined) {
         notifications[index].attributes = {READ : true};
       } else {
         notifications[index].attributes.READ = true;
       }
-      store.count--;
+      $scope.count--;
     }
+
+    var init = function(){
+      $scope.notifications = [];
+      $scope.count = 0;
+      $scope.isEmpty = false;
+      $scope.notificationUrl = NOTIFICATION.notificationFullURL;
+
+      if(NOTIFICATION.groupFiltering) {
```

Comment:
Might be good to default to filtering by group if `NOTIFICATION.groupFiltering` is not set, so as to fail safe on ambiguous configuration.